consensus/consortium/v2/snapshot.go

```diff
@@ -220,6 +220,26 @@ func (s *Snapshot) inturn(validator common.Address) bool {
 	return validators[offset] == validator
 }
 
+// inturn returns if a validator at a given block height is in-turn or not.
+func (s *Snapshot) distanceToInturn(validator common.Address) int {
+	validators := s.validators()
+
+	validatorIndex := 0
+	for i, val := range validators {
+		if val == validator {
+			validatorIndex = i
+			break
+		}
+	}
+
+	offset := (s.Number + 1) % uint64(len(validators))
+	if validatorIndex > int(offset) {
```

Comment:
`validatorIndex == offset` in case validator is inturn, which returns earlier in `backOffTime`, so this case doesn't happen.
But I see the problem of this code is that in case the validator is not in validators list, `validatorIndex == 0`, so `validatorIndex` can equal to `offset` if `offset == 0` too.